libraries/aws_elb.rb

```diff
@@ -14,53 +14,71 @@ class AwsElb < AwsResourceBase
-      policies_in_use     = resp.listener_descriptions.map(&:policy_names).flatten.uniq
-      @ssl_policies       = @aws.elb_client.describe_load_balancer_policies(load_balancer_name: opts[:load_balancer_name])
-                                .policy_descriptions.select { |p| policies_in_use.include?(p.policy_name) }
-    end
+    @availability_zones = load_balancer_description.availability_zones
+    @dns_name           = load_balancer_description.dns_name
+    @load_balancer_name = load_balancer_description.load_balancer_name
+    @external_ports     = load_balancer_description.listener_descriptions.map { |l| l.listener.load_balancer_port }
+    @instance_ids       = load_balancer_description.instances.map(&:instance_id)
+    @internal_ports     = load_balancer_description.listener_descriptions.map { |l| l.listener.instance_port }
+    @security_group_ids = load_balancer_description.security_groups
+    @subnet_ids         = load_balancer_description.subnets
+    @vpc_id             = load_balancer_description.vpc_id
+    @listeners          = load_balancer_description.listener_descriptions
+    @protocols          = load_balancer_description.listener_descriptions.map { |l| l.listener.protocol }.uniq
+    @certificate_id     = load_balancer_description.listener_descriptions.map { |l| l.listener.ssl_certificate_id }
   end
 
   def exists?
-    !@vpc_id.nil?
+    !vpc_id.nil?
   end
 
   def to_s
     "AWS ELB #{load_balancer_name}"
   end
 
   def cross_zone_load_balancing_enabled?
-    !@attrs.cross_zone_load_balancing.enabled == false
+    !!load_balancer_attributes.cross_zone_load_balancing.enabled
   end
 
   def access_log_enabled?
-    !@attrs.access_log.enabled == false
+    !!load_balancer_attributes.access_log.enabled
+  end
+
+  def ssl_policies
+    @ssl_policies ||= catch_aws_errors do
+      elb_client.describe_load_balancer_policies(load_balancer_name: opts[:load_balancer_name])
+                .policy_descriptions.select { |p| policies_in_use.include?(p.policy_name) }
+    end
+  end
+
+  private
+
+  def load_balancer_description
+    @load_balancer_description ||= catch_aws_errors do
+      elb_client.describe_load_balancers(load_balancer_names: [opts[:load_balancer_name]]).load_balancer_descriptions.first
+    end
+  end
+
+  def load_balancer_attributes
+    @load_balancer_attributes ||= catch_aws_errors do
+      elb_client.describe_load_balancer_attributes(load_balancer_name: opts[:load_balancer_name]).load_balancer_attributes
+    end
+  end
+
+  # The ELB will list multiple custom policies, including previously configured policies
+  # Even if a pre-defined policy is selected, a custom policy is created from that template
+  # So, we need to get a list of all policies currently in use and filter the list of policies that exist
+  def policies_in_use
+    load_balancer_description.listener_descriptions.flat_map(&:policy_names).uniq
+  end
+
+  def elb_client
+    @aws.elb_client
```

Comment:
Why we are not using the aws-sdk-elasticloadbalancingv2 sdk instead of aws-sdk-elasticloadbalancing sdk?